packages/engine-server/src/workspace.ts

```diff
@@ -425,39 +451,49 @@ export class WorkspaceService {
     const out = await Promise.all(
       allRepos.map(async (root) => {
         const git = new Git({ localUrl: root });
-        if (await git.hasRemote()) {
+        // It's impossible to pull if there is no remote, or if there are tracked files that have changes
+        if (
+          !(await git.hasRemote()) ||
+          (await git.hasChanges({ untrackedFiles: "no" }))
+        )
+          return undefined;
```

Comment:
do we return some sort of error or warning on this?

Reply:
Revised the reporting so the Sync command is more informative. Please see `Sync.ts`.